packages/commonwealth/server/controllers/server_groups_methods/update_group.ts

```diff
@@ -118,53 +134,15 @@ export async function __updateGroup(
     // update group
     await group.update(toUpdate, { transaction });
 
-    if (topicsToAssociate) {
-      // add group to all specified topics
-      await this.models.Topic.update(
-        {
-          group_ids: sequelize.fn(
-            'array_append',
-            sequelize.col('group_ids'),
-            group.id,
-          ),
-        },
-        {
-          // @ts-expect-error StrictNullChecks
-          where: {
-            id: {
-              [Op.in]: topicsToAssociate.map(({ id }) => id),
-            },
-            [Op.not]: {
-              group_ids: {
-                [Op.contains]: [group.id],
-              },
-            },
-          },
-          transaction,
-        },
-      );
-
-      // remove group from existing group topics
-      await this.models.Topic.update(
-        {
-          group_ids: sequelize.fn(
-            'array_remove',
-            sequelize.col('group_ids'),
-            group.id,
-          ),
-        },
-        {
-          // @ts-expect-error StrictNullChecks
-          where: {
-            id: {
-              [Op.notIn]: topicsToAssociate.map(({ id }) => id),
-            },
-            group_ids: {
-              [Op.contains]: [group.id],
-            },
-          },
-          transaction,
-        },
+    // if associated with some topics, update the groupPermissions
+    if (topics) {
+      group.groupPermissions = await updateGroupPermissions(
+        topics,
+        allowedActions ??
+          (Object.values(ForumActionsEnum) as unknown as ForumActions),
```

Comment:
`ForumActions` type is a single enum property. This is setting an array of values to be equivalent to a single value. If the function should handle array values then the function type should be updated.

Reply:
Good catch